### PR TITLE
deprecate children, migrate picker props

### PR DESCRIPTION
**Pull request by adids1221**
## Description
Picker children, migrate prop deprecation.
As part of the Picker refactor we are deprecating the `children, migrate, onShow` props and moving to use the items prop.
Code cleanup from `migrate, children, getItemLabel, getItemValue, onShow` will be done in another PR.

## Changelog
Picker children, migrate prop deprecation.

## Additional info
MADS-4180

### CODE REVIEW

File: src/components/picker/helpers/usePickerMigrationWarnings.tsx [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
-    if (mode === PickerModes.SINGLE && Array.isArray(value)) {
-      LogService.warn('Picker in SINGLE mode cannot accept an array for value');
+    if (children) {
+      LogService.warn(`UILib Picker will stop supporting the 'children' prop in the next major version, please pass 'items' prop instead`);
     }
-    if (mode === PickerModes.MULTI && !Array.isArray(value)) {
-      LogService.warn('Picker in MULTI mode must accept an array for value');
+
+    if (migrate) {
+      LogService.warn(`UILib Picker will stop supporting the 'migrate' prop in the next major version, please stop using it`);
```
Comment (Inbal-Tish) [inline src/components/picker/helpers/usePickerMigrationWarnings.tsx:16]: Please explain why 'migrate' is no longer neccessary

File: src/components/picker/helpers/usePickerMigrationWarnings.tsx [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
     }
 
-    if (_.isPlainObject(value)) {
-      LogService.warn('UILib Picker will stop supporting passing object as value in the next major version. Please use either string or a number as value');
+    if (getItemLabel) {
+      LogService.warn(`UILib Picker will stop supporting the 'getItemLabel' prop in the next major version, please stop using it`);
```
Comment (Inbal-Tish) [inline src/components/picker/helpers/usePickerMigrationWarnings.tsx:20]: What will replace it?

File: src/components/picker/helpers/usePickerMigrationWarnings.tsx [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
     }
 
-    if (useNativePicker) {
-      LogService.warn(`UILib Picker will stop supporting the 'useNativePicker' prop soon, please pass instead the 'useWheelPicker' prop and handle relevant TextField migration if required to`);
+    if (getItemValue) {
+      LogService.warn(`UILib Picker will stop supporting the 'getItemValue' prop in the next major version, please stop using it`);
```
Comment (Inbal-Tish) [inline src/components/picker/helpers/usePickerMigrationWarnings.tsx:24]: What will replace it?

**Comment by Inbal-Tish** [comment]
@adids1221 Can we remove all the deprecated props that are already commented out?

**Comment by adids1221** [comment]
> @adids1221 Can we remove all the deprecated props that are already commented out?

yes, I'll add another prop to the deprecation `onShow`.

**Review by Inbal-Tish — APPROVED** [review]

**Pull request merged**